packages/mcfunction/src/parser/macro.ts

```diff
@@ -0,0 +1,101 @@
+import * as core from '@spyglassmc/core'
+import { localeQuote, localize } from '@spyglassmc/locales'
+import type { MacroNode } from '../node/index.js'
+
+/**
+ * Parse a macro line.
+ */
+export function macro(): core.Parser<MacroNode> {
+	return (
+		src: core.Source,
+		ctx: core.ParserContext,
+	): core.Result<MacroNode> => {
+		const ans: MacroNode = {
+			type: 'mcfunction:macro',
+			range: core.Range.create(src.cursor),
+			children: [],
+			path: [],
```

Comment:
What is `path` for?

Reply:
I was also wondering this. I just tried removing them and the tests still pass and it still works when trying it out locally.

Reply:
It was to avoid this error we were getting whenever a character was changed
`[Error - 12:32:06 AM] [Service] [getSignatureHelp] Failed for “file:///c:/Users/mulve/AppData/Roaming/PrismLauncher/instances/1.21/.minecraft/saves/New%20World/datapacks/enchantment_reference/data/funny_mobs/functions/funny_mob/spawn.mcfunction” #38 TypeError: Cannot read properties of undefined (reading 'length')
    at getOptions3 (c:\Users\mulve\Documents\misode-spyglass\packages\vscode-extension\dist\server.js:46280:54)
    at c:\Users\mulve\Documents\misode-spyglass\packages\vscode-extension\dist\server.js:46237:21
    at Service.getSignatureHelp (c:\Users\mulve\Documents\misode-spyglass\packages\vscode-extension\dist\server.js:38357:24)
    at c:\Users\mulve\Documents\misode-spyglass\packages\vscode-extension\dist\server.js:47801:24
[Error - 12:32:36 AM] [Service] [getSignatureHelp] Failed for “file:///c:/Users/mulve/AppData/Roaming/PrismLauncher/instances/1.21/.minecraft/saves/New%20World/datapacks/enchantment_reference/data/funny_mobs/functions/funny_mob/spawn.mcfunction” #71 TypeError: Cannot read properties of undefined (reading 'length')`
I traced it down to the path not existing for the nodes

Reply:
Can we fix the mcfunction signature provider instead of adding an unnecessary `path` property to macro nodes?

Reply:
Specifically the type assertion here is no longer safe since macro nodes can also be selected here. (Why didn't comment nodes cause us this problem?)

https://github.com/SpyglassMC/Spyglass/blob/9d415a9b8d06e9367fa2e0434f291aeae8e57766/packages/java-edition/src/mcfunction/signatureHelpProvider.ts#L70-L72

Reply:
I see, the issue is that `java-edition/src/mcfunction/signatureHelpProvider.ts` `getSelectedCommandNode` is incorrectly assuming that every child it finds will be a `CommandNode` (by casting to it). The fix would be to check the node type in `signatureHelpProvider` and only continue when it's actually a command node.

Reply:
NINJA'D

Reply:
So should I remove the path stuff and just leave the bug to be fixed in another PR?